storm-core/src/clj/org/apache/storm/daemon/supervisor.clj

```diff
@@ -332,19 +333,24 @@
    :assignment-id (.getAssignmentId isupervisor)
    :my-hostname (Utils/hostname conf)
    :curr-assignment (atom nil) ;; used for reporting used ports when heartbeating
-   :heartbeat-timer (mk-timer :kill-fn (fn [t]
-                               (log-error t "Error when processing event")
-                               (Utils/exitProcess 20 "Error when processing an event")
-                               ))
-   :event-timer (mk-timer :kill-fn (fn [t]
-                                         (log-error t "Error when processing event")
-                                         (Utils/exitProcess 20 "Error when processing an event")
-                                         ))
-   :blob-update-timer (mk-timer :kill-fn (defn blob-update-timer
-                                           [t]
-                                           (log-error t "Error when processing event")
-                                           (Utils/exitProcess 20 "Error when processing a event"))
-                                :timer-name "blob-update-timer")
+   :heartbeat-timer (StormTimer/mkTimer nil
+                      (reify StormTimer$TimerFunc
```

Comment:
This implementation of TimerFunc is repeated in many places. Does it make sense to move it to java based implementation of TimerFunc and use everywhere?